tests/test_structures/test_bbox/test_horizontal_bbox.py

```diff
@@ -0,0 +1,166 @@
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        bboxes = HorizontalBoxes(th_bboxes)
+
+        # Centers
+        centers = torch.Tensor([15, 15]).reshape(1, 1, 2)
+        assert_allclose(bboxes.centers, centers)
+        # Areas
+        areas = torch.Tensor([100]).reshape(1, 1)
+        assert_allclose(bboxes.areas, areas)
+        # widths
+        widths = torch.Tensor([10]).reshape(1, 1)
+        assert_allclose(bboxes.widths, widths)
+        # heights
+        heights = torch.Tensor([10]).reshape(1, 1)
+        assert_allclose(bboxes.heights, heights)
+
+    def test_flip(self):
+        img_shape = [50, 85]
+        # horizontal flip
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        flipped_bboxes_th = torch.Tensor([65, 10, 75, 20]).reshape(1, 1, 4)
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.flip_(img_shape, direction='horizontal')
+        assert_allclose(bboxes.tensor, flipped_bboxes_th)
+        # vertical flip
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        flipped_bboxes_th = torch.Tensor([10, 30, 20, 40]).reshape(1, 1, 4)
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.flip_(img_shape, direction='vertical')
+        assert_allclose(bboxes.tensor, flipped_bboxes_th)
+        # diagonal flip
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        flipped_bboxes_th = torch.Tensor([65, 30, 75, 40]).reshape(1, 1, 4)
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.flip_(img_shape, direction='diagonal')
+        assert_allclose(bboxes.tensor, flipped_bboxes_th)
+
+    def test_translate(self):
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.translate_([23, 46])
+        translated_bboxes_th = torch.Tensor([33, 56, 43, 66]).reshape(1, 1, 4)
+        assert_allclose(bboxes.tensor, translated_bboxes_th)
+
+    def test_clip(self):
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        img_shape = [13, 14]
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.clip_(img_shape)
+        cliped_bboxes_th = torch.Tensor([10, 10, 14, 13]).reshape(1, 1, 4)
+        assert_allclose(bboxes.tensor, cliped_bboxes_th)
+
+    def test_rotate(self):
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        center = (15, 15)
+        angle = 45
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.rotate_(center, angle)
+        rotated_bboxes_th = torch.Tensor([
+            15 - 5 * sqrt(2), 15 - 5 * sqrt(2), 15 + 5 * sqrt(2),
+            15 + 5 * sqrt(2)
+        ]).reshape(1, 1, 4)
+        assert_allclose(bboxes.tensor, rotated_bboxes_th)
+
+    def test_project(self):
+        th_bboxes = torch.Tensor([10, 10, 20, 20]).reshape(1, 1, 4)
+        matrix = np.random.rand(3, 3)
+        bboxes = HorizontalBoxes(th_bboxes)
+        bboxes.project_(matrix)
```

Comment:
consider using a speical case to test whether return target value.

Reply:
I use rotation, translation, and scaling to build the matrix. Test if the project_ is equal to the combination of rotation, translation and scaling.